Bottom: aca5ed0f7c6c8b31c7626f92e72dfef531d38271
Top:    fe857cb32e584e64908991a3c8f65184c10acff4
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2011-12-25 00:46:57 +0000

hid/gtk: Avoid warnings when netlist window is not created.

1. When the NetlistChanged action is fired, we should only update if the
   netlist window is visible. (It will be created up to date when the
   user asks for it).

2. Always show the netlist window when requested, even if it has no
   entries. This is good for consistent UI behaviour anyway, as the user
   should be shown a blank netlist window if they request it.. even if
   the is no content to display.


---

diff --git a/src/hid/gtk/gui-netlist-window.c b/src/hid/gtk/gui-netlist-window.c
index 249066f..e2c08d5 100644
--- a/src/hid/gtk/gui-netlist-window.c
+++ b/src/hid/gtk/gui-netlist-window.c
@@ -699,11 +699,6 @@ ghid_netlist_window_create (GHidPort * out)
   GtkCellRenderer *renderer;
   GtkTreeViewColumn *column;
 
-  /* No point in putting up the window if no netlist is loaded.
-   */
-  if (!PCB->NetlistLib.MenuN)
-    return;
-
   if (netlist_window)
     return;
 
@@ -1014,6 +1009,10 @@ GhidNetlistChanged (int argc, char **argv, Coord x, Coord y)
   if (ghidgui == NULL || !ghidgui->is_up)
     return 0;
 
+  /* There is no need to update if the netlist window isn't open */
+  if (netlist_window == NULL)
+    return 0;
+
   loading_new_netlist = TRUE;
   ghid_netlist_window_update (TRUE);
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (disable_all_button),
